core/src/main/kotlin/supplier/CacheAwareRestSupplier.kt

```diff
@@ -0,0 +1,171 @@
+package dev.kord.core.supplier
+
+import dev.kord.cache.api.DataCache
+import dev.kord.cache.api.put
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.entity.*
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.GuildChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+/**
+ * [EntitySupplier] that uses a [RestEntitySupplier] to resolve entities.
+ *
+ * Resolved entities will always be stored in [cache] if it wasn't null or empty for flows.
+ */
+class CacheAwareRestEntitySupplier(
+    val restSupplier: RestEntitySupplier,
+    val cache: DataCache,
+    val kord: Kord
```

Comment:
These properties should probably be private, I don't see anyone interaction with them when using a supplier